service/orchestrator/assessment_results.go

```diff
@@ -27,48 +27,71 @@ package orchestrator
 
 import (
 	"context"
+	"fmt"
 
 	"golang.org/x/exp/maps"
 	"golang.org/x/exp/slices"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
 	"clouditor.io/clouditor/api/assessment"
+	"clouditor.io/clouditor/internal/util"
 	"clouditor.io/clouditor/service"
 )
 
 // ListAssessmentResults is a method implementation of the orchestrator interface
 func (svc *Service) ListAssessmentResults(ctx context.Context, req *assessment.ListAssessmentResultsRequest) (res *assessment.ListAssessmentResultsResponse, err error) {
 	var values = maps.Values(svc.results)
-	var filtered []*assessment.AssessmentResult
+	var filtered_values []*assessment.AssessmentResult
 	var allowed []string
 	var all bool
 
+	// Validate request
+	err = req.Validate()
+	if err != nil {
+		err = fmt.Errorf("invalid request: %v", err)
+		log.Error(err)
+		return nil, status.Errorf(codes.InvalidArgument, "%s", err)
+	}
+
+	filteredCloudServiceId := util.Deref(req.FilteredCloudServiceId)
+
 	// Check, if the current filter is valid according to the authorization strategy. Omitting the cloud service ID is
 	// only allowed, if one can access *all* the cloud services.
 	all, allowed = svc.authz.AllowedCloudServices(ctx)
-	if !all && req.FilteredCloudServiceId == "" {
+	if !all && filteredCloudServiceId == "" {
```

Comment:
Shouln't  we check for nil instead? And we can use the `GetXXX` functions of the protobuf message, this removes the need for the `Derek` call.